tools/Microsoft.AspNet.Server.Kestrel.GeneratedCode/FrameFeatureCollection.cs

```diff
@@ -25,26 +25,40 @@ public virtual void BeforeCompile(BeforeCompileContext context)
 
         public static string GeneratedFile()
         {
-            var commonFeatures = new[]
+            var alwaysFeatures = new[]
             {
                 typeof(IHttpRequestFeature),
                 typeof(IHttpResponseFeature),
                 typeof(IHttpRequestIdentifierFeature),
-                typeof(IHttpSendFileFeature),
                 typeof(IServiceProvidersFeature),
-                typeof(IHttpAuthenticationFeature),
                 typeof(IHttpRequestLifetimeFeature),
+                typeof(IHttpConnectionFeature)
+            };
+
+            var commonFeatures = new[]
+            {
+                typeof(IHttpAuthenticationFeature),
                 typeof(IQueryFeature),
-                typeof(IFormFeature),
+                typeof(IFormFeature)
+            };
+
+            var sometimesFeatures = new[]
+            {
+                typeof(IHttpUpgradeFeature),
```

Comment:
@tratcher the fact they are less common or rare has no bearing. The collection has interfaces that are known in advance which is one less reason to create the list in any series of requests. Just like well-known headers in iis/httpsys structs. Those don't decide that headers which virtually never occur don't get an intptr's worth of memory to represent them.

The only thing we need to know is what feature types you can reach from our systems, to avoid creating the extra collection, and which are provided by the server, to avoid the 'as' casting from a known type.

I'm responding from a phone so it's a bit difficult to tell. Is that the state of this PR's implementation?